server/server.js

```diff
@@ -26,6 +30,194 @@ app.get('/api/hello', (req, res) => {
   res.json({ message: 'Hello World!' });
 });
 
+app.post('/api/users/register', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) { throw new ClientError(400, 'username and password are required fields'); }
+    const hashedPassword = await argon2.hash(password);
+    const date = new Date();
+    const sql = `
+      insert into "users" ("username", "hashedPassword", "createdAt", "loggedInAt")
+      values ($1, $2, $3, $4)
+      returning *
+    `;
+    const params = [username, hashedPassword, date, date];
+    const result = await db.query(sql, params);
+    const user = result.rows[0];
+    res.status(201).json(user);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/users/log-in', async (req, res, next) => {
+  try {
+    const { username, password } = req.body;
+    if (!username || !password) { throw new ClientError(401, 'Invalid Login'); }
+    const sql = `
+    select "userId",
+     "hashedPassword"
+      from "users"
+      where "username" = $1
+    `;
+    const params = [username];
+    const result = await db.query(sql, params);
+    if (!result) { throw new ClientError(401, 'Invalid Login'); }
+    const user = result.rows[0];
+    console.log('log-in');
+    if (!user) { throw new ClientError(401, 'Invalid Login'); }
+    const { userId, hashedPassword } = user;
+    if (!await argon2.verify(hashedPassword, password)) { throw new ClientError(401, 'Invalid Login'); }
+    const payload = { userId, user };
+    const token = jwt.sign(payload, process.env.TOKEN_SECRET);
+    res.json({ user: payload, token });
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.use(authorizationMiddleware);
+
+app.post('/api/images/upload', async (req, res, next) => {
```

Comment:
For this to work, you need to include the uploads-middleware. See the react-file-uploads exercise.